### PR TITLE
Screencopy Protocol Support

### LINKED ISSUE #262: Screencopy Protocol
While screenshooter is a nice protocol, screencopy is a newer refresh and wlroots team supports it. [Grim](https://github.com/emersion/grim) supports screencopy only. I'm not aware of many programs that use screenshooter.

[Screencopy Protocol](https://github.com/swaywm/wlroots/blob/master/protocol/wlr-screencopy-unstable-v1.xml)
[Wlroots Header](https://github.com/swaywm/wlroots/blob/master/include/wlr/types/wlr_screencopy_v1.h)

If no one else beats me to it, I'll see if I can knock it out this weekend.

### PR DESCRIPTION
**Pull request by SethBarberee**
Closes #262 

* Had to dig a little deeper and import the header file
* In Build.rs, I tried to make it look a little cleaner. I'll take a look at the macro you described when I can.

I'm planning to continue doing this. Protocols are fun :)

### CODE REVIEW

File: src/extensions/screencopy.rs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+//! Support for the wlroots Screencopy (Version 1) Protocol
+
+use wayland_sys::server::wl_display as wl_server_display;
+use wlroots_sys::{wl_display, wlr_screencopy_manager_v1, wlr_screencopy_manager_v1_create,
+    wlr_screencopy_manager_v1_destroy};
+
+#[derive(Debug)]
+/// Manager that offers requests to start capturing from a source
+pub struct Manager {
+    manager: *mut wlr_screencopy_manager_v1
```
Comment (Timidger) [inline src/extensions/screencopy.rs]: This protocol is unstable (due to the `z` in its name and the fact it has a "v1" in it). It's possible there are going to be many versions of this that will need to be supported as the final decision is made on the stabilization. This happened with xdg-shell (v5 and v6 existed for a while, and now _most_ use the stable version).

So far the way I've encoded this is by adding a `V#` in the type somewhere, like I did with xdg shell. So we should name this `ManagerV1` or even better `ZManagerV1`. 

You should also point out in the docs that this is unstable and you should link to the protocol (I've been flubbing on the documentation thus far, so I won't hold against you if you do the same).

We cannot use a feature flag, as different versions of the same protocol can exist at the same time.

Comment (SethBarberee) [reply src/extensions/screencopy.rs]: Gotcha. I'll get this fixed up later today. Thanks for the feedback. I'll keep it in mind with the other unstable protocols.

**Comment by SethBarberee** [comment]
Fixed up what you suggested. If all is good, I can squash the commits if you want me to.

**Comment by Timidger** [comment]
Looks good, I don't mind squashing or not (it's up to you). If it passes I'll merge it in.

**Comment by SethBarberee** [comment]
If you want to squash them, that's fine with me!

**Comment by Timidger** [comment]
Thanks!

**Pull request merged**